### ISSUE TITLE
Converters do not work with `IOSvc`

### ISSUE THREAD

**Issue by tmadlener** [key4hep/k4MarlinWrapper#202]
The converters still partially refer to the `PodioDataSvc` explicitly and will fail with a segmentation violation when used with the `IOSvc`.

**Comment by tmadlener** [comment]
This probably needs a bit more work, because currently the `PodioDataSvc` is pretty ingrained in some places, and for some things I am not sure if they will work with the `IOSvc`.

The `LcioEventAlgo` currently puts an `LCEventWrapper` into the `eventSvc`
https://github.com/key4hep/k4MarlinWrapper/blob/3165e403823a2f6bc80091cc4cef94a1d99f8d2f/k4MarlinWrapper/src/components/LcioEventAlgo.cpp#L67-L68
Does this still work with the `IOSvc`? @jmcarcell IIUC, this should still work, but the `LCEvent` will not be available for our functional algorithms, right?

For the converters it should be possible to put in place checks that check the `IOSvc`, resp. `MetadataSvc` and the `PodioDataSvc`.

**Comment by tmadlener** [comment]
As discussed during the [EDM4hep meeting](https://indico.cern.ch/event/1459928/), we will not put too much effort into supporting `IOSvc` with the k4MarlinWrapper, because that is intended mainly for multithreaded usage, and only very few Marlin processors are actually thread safe. We will keep this issue open, so that it is visible for people who try to use the `IOSvc` with the wrapper. 

If necessary we will come back to this in the future to add support.

**Comment by BrieucF** [comment]
Trying to raise the priority of this issue: it is not really a matter of running in multi-threaded way, it is rather that we try to write all the new algorithms with Functionals and this issue currently prevents one to access the ILCSoft world.

**Comment by samf25** [comment]
I would also like to add in my voice, here. The fact that we can't use the converters with Transformers is hampering my progress on Muon Collider work. I've converted a number of Marlin Processors to Transformers but now I have to run the converted algorithms separately from the ones that are still in MarlinWrappers.

**Comment by BrieucF** [comment]
Hi @samf25 , are these translated Marlin Processors available centrally in Key4hep?

**Comment by samf25** [comment]
No. Currently, they are just located in a [separate repo](https://github.com/samf25/ACTSTracking/tree/multiK4FWC) (there's also [TrackPerf](https://github.com/samf25/TrackPerf/tree/switchK4FWC), and I am currently working on some [digitization](https://github.com/spg-berkeleylab/MarlinTrkProcessors/tree/K4FWC) [libraries](https://github.com/spg-berkeleylab/MuonCVXDDigitiser/tree/K4FWC)). Once I can get it all hooked up and running (with MarlinWrappers around they as of yet untranslated ones) they will be added to the MuCol software stack. I don't think there are any plans to include them centrally since they are specific to the muon collider.
